src/fontra/views/editor/editor.js

```diff
@@ -2435,8 +2435,16 @@ export class EditorController {
     const hasAnchors = instance.anchors.length > 0;
     const hasGuidelines = instance.guidelines.length > 0;
 
+    const glyphPath = positionedGlyph.glyph.path;
+    let onCurvePoints = [];
```

Comment:
It would be more correct if we'd check: "does the selection contain all on-curve points?"

(Imagine a path with four on-curve points and four off-curve points. Select to on-curve and two off-curve, and your current logic alreade considers this "all points have been selected".)

I suggest to use a Set for onCurvePoints (replace onCurvePoints.push with onCurvePoints.add), then use `isSuperset(onCurvePoints, pointIndices)` to get the answer to the above question.

isSuperset comes from set-ops.js, and the second argument does not need to be a set: you can use pointIndices as-is.

Reply:
Compute isSuperset() only once: you use its inverse below. As least that's how I read your code.